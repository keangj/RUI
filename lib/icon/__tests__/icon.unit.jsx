import * as renderer from 'react-test-renderer';
import React from 'react';
import Icon from '../icon';
import { mount } from 'enzyme'

describe('icon', () => {
  it('icon 渲染成功', () => {
    const json = renderer.create(<Icon name="qq"/>).toJSON();
    expect(json).toMatchSnapshot();  // 在 __snapshot__ 下生成快照，用 yarn test -u 更新快照
  });
  it('onClick', () => {
    const fn = jest.fn();
    const component = mount(<Icon name="qq" onClick={fn}/>);
    component.find('svg').simulate('click');
    expect(fn).toBeCalled();
  })
});