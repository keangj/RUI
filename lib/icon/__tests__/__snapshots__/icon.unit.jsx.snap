// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon icon 渲染成功 1`] = `
<svg
  className="rui-icon"
>
  <use
    xlinkHref="#qq"
  />
</svg>
`;

exports[`icon 是个 div 1`] = `
<svg
  className="rui-icon"
>
  <use
    xlinkHref="#qq"
  />
</svg>
`;
